ts/components/conversation/header/ConversationHeader.tsx

```diff
@@ -57,3 +72,71 @@ export const ConversationHeaderWithDetails = () => {
     </div>
   );
 };
+
+const RecreateGroupContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-self: center;
+  width: 100%;
+
+  .session-button {
+    padding-inline: var(--margins-3xl);
+  }
+`;
+
+function useShowRecreateModal() {
+  const dispatch = useDispatch();
+
+  return useCallback(
+    (name: string, members: Array<PubkeyType>) => {
+      dispatch(
+        updateConfirmModal({
+          title: localize('recreateGroup').toString(),
+          i18nMessage: { token: 'legacyGroupChatHistory' },
+          okText: localize('theContinue').toString(),
+          cancelText: localize('cancel').toString(),
+          okTheme: SessionButtonColor.Danger,
+          onClickOk: () => {
+            dispatch(setLeftOverlayMode('closed-group'));
+            dispatch(groupInfoActions.updateGroupCreationName({ name }));
+            dispatch(groupInfoActions.setSelectedGroupMembers({ membersToSet: members }));
+          },
+          onClickClose: () => {
+            dispatch(updateConfirmModal(null));
+          },
+        })
+      );
+    },
+    [dispatch]
+  );
+}
+
+function RecreateGroupButton() {
+  const isLegacyGroup = useSelectedIsLegacyGroup();
+  const selectedConvo = useSelectedConversationKey();
+
+  const name = useConversationUsername(selectedConvo);
+  const members = use05GroupMembers(selectedConvo);
+
+  const weAreAdmin = useSelectedWeAreAdmin();
+
+  const showRecreateGroupModal = useShowRecreateModal();
+
+  if (!isLegacyGroup || !weAreAdmin) {
+    return null;
+  }
+
+  return (
+    <RecreateGroupContainer>
+      <SessionButton
+        buttonColor={SessionButtonColor.Primary}
+        margin="var(--margins-sm)"
+        onClick={() => {
+          showRecreateGroupModal(name || 'Unknown group name', members);
```

Comment:
Should `Unknown group name` not be localised?

Reply:
made it to  be "unknown group" localized